src/nvim/mouse.c

```diff
@@ -212,22 +212,30 @@ static int get_fpos_of_mouse(pos_T *mpos)
   if (wp == NULL) {
     return IN_UNKNOWN;
   }
+  int winrow = row;
+
+  // compute the position in the buffer line from the posn on the screen
+  bool below_buffer = mouse_comp_pos(wp, &row, &col, &mpos->lnum);
+
+  if (!below_buffer && *wp->w_p_stc != NUL && mouse_col < win_col_off(wp)) {
+    return MOUSE_STATUSCOL;
+  }
 
   // winpos and height may change in win_enter()!
-  if (row + wp->w_winbar_height >= wp->w_height) {  // In (or below) status line
+  if (mouse_row + wp->w_winbar_height >= wp->w_height) {  // In (or below) status line
```

Comment:
Why is `mouse_row` used here to compare with window height? This doesn't make sense.

Reply:
Mistake. I moved the `mouse_comp_pos()` call up to use its result in the statuscolumn comparison. After which the original `row/col` are no longer valid for the other comparisons.
Fixed now to use the intermediate `winrow/col`.